portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +37,53 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Neel!</h1>");
+    
+    // create query to order comments by date from data store
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // convert to array list of comments
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+    for (Entity entity : results.asIterable()) {
+      String username = (String) entity.getProperty("username");
```

Comment:
I see that makes sense. So then for the Date object (the following line, not shown) typecasting is also not necessarily safe. How am I supposed to be able to convert it to a date object as needed to construct the comment object? Or for any other non-string fields that I may add in the future?

Reply:
In the case with Date you don't really have a choice because there's no good way to do so.  String is the one exception since the method exists.  Checking the type at runtime is an option but it's so slow that it's best to be saved for situations where you have to be absolutely sure.  For now I think it's enough to try to catch these errors with testing.